features/behat-steps.feature

```diff
@@ -0,0 +1,741 @@
+      """
+
+  Scenario: Test multiple files in directory
+    Given an empty directory
+    And a file-a.txt file:
+      """
+      A
+      """
+    And a file-b.txt file:
+      """
+      B
+      """
+    And a file-c.txt file:
+      """
+      C
+      """
+    Then the {RUN_DIR} directory should contain:
+      """
+      file-a.txt
+      """
+    And the {RUN_DIR} directory should contain:
+      """
+      file-b.txt
+      """
+    And the {RUN_DIR} directory should contain:
+      """
+      file-c.txt
+      """
+
+  Scenario: Test special characters in file content
+    Given an empty directory
+    And a special.txt file:
+      """
+      Line with "quotes"
+      Line with 'apostrophes'
+      Line with $variable
+      """
+    Then the special.txt file should contain:
+      """
+      quotes
+      """
+
+  Scenario: Test version comparison operators
+    When I run `echo "5.6.2"`
+    Then STDOUT should be a version string > 5.6.1
+    And STDOUT should be a version string >= 5.6.2
+    And STDOUT should be a version string < 5.6.3
+    And STDOUT should be a version string <= 5.6.2
+    And STDOUT should be a version string == 5.6.2
+    And STDOUT should be a version string != 5.6.3
+
+  @require-wp
+  Scenario: Test JSON array containing
+    Given a WP installation
+    When I run `wp eval 'echo json_encode(["apple", "banana", "cherry"]);'`
+    Then STDOUT should be a JSON array containing:
+      """
+      ["apple", "banana"]
+      """
+
+  @require-wp
+  Scenario: Test email sending detection
+    Given a WP installation
+    When I run `wp user create testuser test@example.com --role=subscriber`
+    Then an email should not be sent
+
+  @require-wp
+  Scenario: Test STDOUT end with table
+    Given a WP installation
+    When I run `wp user list --fields=user_login --format=table`
+    Then STDOUT should end with a table containing rows:
+      | user_login |
+      | admin      |
+
+  Scenario: Test combining multiple string replacements
+    Given an empty directory
+    And a template.txt file:
+      """
+      Hello FIRST_NAME LAST_NAME
+      """
+    And "FIRST_NAME" replaced with "John" in the template.txt file
+    And "LAST_NAME" replaced with "Doe" in the template.txt file
+    Then the template.txt file should contain:
+      """
+      Hello John Doe
+      """
+
+  Scenario: Test nested directory creation
+    Given an empty directory
+    And a deep/nested/path/file.txt file:
+      """
+      Deep content
+      """
+    Then the deep/nested/path/file.txt file should exist
+
+  Scenario: Test absolute vs relative paths
+    Given an empty directory
+    And a relative.txt file:
+      """
+      Relative
+      """
+    Then the relative.txt file should exist
+    And the {RUN_DIR}/relative.txt file should exist
+
+  @require-phar
+  Scenario: Test new Phar with specific version
+    Given an empty directory
+    And a new Phar with version "2.11.0"
+    Then the wp-cli.phar file should exist
```

Comment:
I think we actually have to remove this test as Phar building only works when wp-cli-bundle is available, which isn't the case here.